app/v2/notifications/post_notifications.py

```diff
@@ -47,6 +50,70 @@
 from app.utils import get_public_notify_type_text
 
 
+def wrap_recipient_identifier_in_pii(form: dict) -> dict:
+    """Wrap the recipient identifier id_value in the appropriate PII class.
+
+    This function takes a form containing recipient identifier data and wraps
+    the id_value in the appropriate PII class based on the id_type. This provides
+    encryption and controlled access to personally identifiable information.
+
+    Args:
+        form (dict): The validated form data containing recipient_identifier.
+                    This dictionary may be modified in-place if a valid
+                    recipient_identifier with id_type and id_value is present.
+
+    Returns:
+        dict: The same form dictionary, potentially with the
+              recipient_identifier['id_value'] wrapped in an appropriate
+              PII class (PiiIcn, PiiEdipi, PiiBirlsid, PiiPid, or PiiVaProfileID).
+              If wrapping fails or required fields are missing, the form
+              is returned unchanged.
+
+    Note:
+        - The form parameter is modified in-place when PII wrapping occurs
+        - Unknown id_types are logged as warnings but don't raise exceptions
+        - PII instantiation errors are logged but don't prevent function completion
+        - Only processes forms that contain recipient_identifier with both id_type and id_value
+    """
+    recipient_identifier: dict | None = form.get('recipient_identifier')
+    if recipient_identifier is None:
+        return form
+
+    id_type: str = recipient_identifier.get('id_type')
+    id_value: str = recipient_identifier.get('id_value')
+
+    if id_type is None or id_value is None:
+        return form
```

Comment:
```suggestion
    # These values must be present to pass validation.
    id_type: str = recipient_identifier['id_type']
    id_value: str = recipient_identifier['id_value']
```

If you don't want to trust the validation, you could catch KeyError and use that to log that a programming error exists.  I think the above is adequate.

Reply:
So we are restricting this method to only be used by the v2/post_notifications route?